doc/nrf/protocols/matter/getting_started/matter_api.rst

```diff
@@ -0,0 +1,258 @@
+
+  This function is only available if the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` Kconfig option is selected.
+
+For more details regarding nRF Connect Matter initialization API, refer to the Doxygen commentary in the :file:`ncs/nrf/samples/matter/common/src/app/matter_init.h` header file.
+
+Event handler API
+=================
+
+The Matter SDK provides a notification scheme based on the public events that are propagated from the Matter stack to the application layer.
+The nRF Connect Matter event handler API provides mechanisms to register and unregister custom functions that handle these events within an application.
+This module also includes a default handler that is used in nRF Connect SDK Matter samples and applications.
+
+The specific Matter events that can be handled in the application are listed in the :file:`ncs/modules/lib/matter/src/include/platform/CHIPDeviceEvent.h` header file.
+The nRF Connect Matter API contains of the following functions that can be used to handle events:
+
+:c:func:`RegisterEventHandler()`:
+  This function is used to register the provided Matter event handler(``EventHandlerFunct``) in a thread-safe manner.
+  It is safe to call this function in the application after the Matter server has already been initialized.
+
+:c:func:`UnregisterEventHandler()`:
+  This function is used to unregister the provided Matter event handler(``EventHandlerFunct``) in a thread-safe manner.
+  It is safe to call this function in the application after the Matter server has already been initialized.
+
+:c:func:`DefaultEventHandler()`:
+  This is an nRF Connect Matter event handler function that is registered in the nRF Connect Matter Initialization API by default.
+  You can unregister this handler with the :c:func:`UnregisterEventHandler()` function in the application if needed.
+
+For more details regarding nRF Connect Matter event handler API, refer to the Doxygen commentary in the :file:`ncs/nrf/samples/matter/common/src/app/matter_event_handler.h` header file.
+
+nRF Connect Matter API usage example
+====================================
+
+Combining both aforementioned nRF Connect Matter APIs, you can develop an application by following the initialization scheme listed below:
+
+.. code-block:: C++
+
+   #include "app/matter_init.h"
+
+   #include <zephyr/logging/log.h>
+
+   LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+   static void CustomMatterEventHandler(const ChipDeviceEvent *event, intptr_t /* unused */)
+   {
+     switch (event->Type) {
+     case DeviceEventType::kCHIPoBLEAdvertisingChange:
+     /* Custom code, e.g. control LED */
+     break;
+     }
+   }
+
+   CHIP_ERROR MatterAppInit()
+   {
+     /* Initialize Matter stack */
+     ReturnErrorOnFailure(Nrf::Matter::PrepareServer(Nrf::Matter::InitData{ .mPostServerInitClbk = [] {
+     LOG_INF("Matter server has been initialized.");
```

Comment:
spaces here and below?

Btw. Not sure what is the standard for code snippets.. but only two spaces seems a bit narrow.

Reply:
@wiba-nordic do we have any standard here? Look like sth that could be changed indeed.

Reply:
I don't think we actually have any standard for code snippets. The assumption is that they come from actual code, and would follow what is used there.